lib/core/lib/ops/validation.js

```diff
@@ -4,14 +4,39 @@ const Joi = require('joi')
 
 const requiredString = Joi.string().required()
 const requiredStringId = Joi.string().required().max(128)
+const MetaData = Joi.object().optional().description('Arbitrary Metadata').label('MetaData')
 
+const PolicyIdObject = Joi.object({
+  id: Joi.string().required().description('Policy Id'),
+  variables: Joi.object().description('A list of the veriables with their fixed values').label('variables')
+}).required().description('Policy Id Object, alternatively supply string instead').label('PolicyIdObject')
+
+const PolicyIdString = Joi.string().required().description('PolicyIdString')
+// it would be ideal to put the policyid object first, however it causes a swagger doc error
+// as hapi-swagger cannot interpret alternatives
 const requiredPolicy = Joi.alternatives().try([
-  requiredString,
-  Joi.object().description('Policy').keys({
-    id: requiredString.description('Policy Id'),
-    variables: Joi.object().description('A list of the veriables with their fixed values')
-  })
-])
+  PolicyIdString,
+  PolicyIdObject
+]).description('Required Policy (can be object or string)').label('RequiredPolicy')
+
+const PolicyIdsArray = Joi.array().required().items(requiredPolicy).description('can be array of objects with id:string, variables:object properties OR an array of string ids').label('PolicyIdsArray')
+const UsersArray = Joi.array().required().items(requiredString).description('User IDs').label('UsersArray')
+const TeamsArray = Joi.array().required().items(requiredString).description('Teams IDs').label('TeamsArray')
+const ResourcesArray = Joi.array().max(10).items(requiredString.description('A single resource')).single().required().description('A list of Resources').label('ResourcesArray')
```

Comment:
why is this limit of 10 added? where does this come from? does pbac impose any limits on this? we should remove this limit

Reply:
max(10) wasn't added, it was declared previously inline on resources: property,

not its declared in specific object to prevent repeating model (see line 41 below)

resources: Joi.array().max(10).items(requiredString.description('A single resource')).single().required().description('A list of Resources'),

Reply:
ps I noticed the limit and thought it was odd, don't know if there is a limit on PBAC, if not it should definitely be removed

Reply:
removing